app/enssubdomain/views.py

```diff
@@ -0,0 +1,119 @@
+# -*- coding: utf-8 -*-
+"""
+    Copyright (C) 2018 Gitcoin Core
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Affero General Public License as published
+    by the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+    GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with this program. If not, see <http://www.gnu.org/licenses/>.
+"""
+
+import datetime
+
+from django.conf import settings
+from django.contrib.auth.decorators import login_required
+from django.http import JsonResponse
+from django.template.response import TemplateResponse
+from django.utils import timezone
+from django.utils.translation import gettext_lazy as _
+from django.views.decorators.csrf import csrf_exempt
+from gas.utils import recommend_min_gas_price_to_confirm_in_time
+
+from dashboard.models import Profile
+from dashboard.views import w3
+from ens import ENS
+from eth_account.messages import defunct_hash_message
+from web3 import HTTPProvider, Web3
+from enssubdomain.web3py.web3.middleware.signing import construct_sign_and_send_raw_middleware
+
+from .models import ENSSubdomainRegistration
+
+ns = ENS.fromWeb3(w3)
+w3 = Web3(HTTPProvider(settings.WEB3_HTTP_PROVIDER))
+ENS_MAINNET_ADDR = '0x314159265dD8dbb310642f98f50C066173C1259b'
+
+
+@csrf_exempt
+@login_required
+def ens_subdomain(request):
+    """Register ENS Subdomain."""
+    github_handle = request.user.profile.handle if request.user.is_authenticated and hasattr(request.user, 'profile') else None
+    if request.method == "POST" and github_handle:
+        signedMsg = request.POST.get('signedMsg', '')
+        signer = request.POST.get('signer', '').lower()
+        if signedMsg and signer:
+            message_hash = defunct_hash_message(text=f'Github Username : {github_handle}')
+            recovered_signer = w3.eth.account.recoverHash(message_hash, signature=signedMsg).lower()
+            if recovered_signer == signer:
+                txn_hash = '0x6477f3640d9d910c589937b25d5892f525cfde2dd634d9490abc7542c946e8e3'
+                transaction = {
+                    'to': ENS_MAINNET_ADDR,
+                    'from': settings.ENS_OWNER_ACCOUNT,
+                    'value': 0,
+                    'nonce': w3.eth.getTransactionCount(settings.ENS_OWNER_ACCOUNT),
+                    'gas': 100000,
+                    'gasPrice': recommend_min_gas_price_to_confirm_in_time(1) * 10**9
+                }
+                #w3.middleware_stack.add(construct_sign_and_send_raw_middleware(settings.ENS_PRIVATE_KEY))
```

Comment:
the middleware should ideally be plugged immediately after  creating a web3 instance on line 42 in the same file.
```python
w3 = Web3(HTTPProvider(settings.WEB3_HTTP_PROVIDER))
w3.middleware_stack.add(construct_sign_and_send_raw_middleware(settings.ENS_PRIVATE_KEY))
```
plugging it inside this view is redundant.